src/main/java/org/opensearch/knn/index/codec/KNN80Codec/KNN80DocValuesConsumer.java

```diff
@@ -184,6 +188,20 @@ private void createKNNIndexFromScratch(FieldInfo fieldInfo, KNNCodecUtil.Pair pa
 
         // Used to determine how many threads to use when indexing
         parameters.put(KNNConstants.INDEX_THREAD_QTY, KNNSettings.state().getSettingValue(KNNSettings.KNN_ALGO_PARAM_INDEX_THREAD_QTY));
+        Object name = parameters.get(NAME);
+        if (name != null && name.equals(METHOD_NSG)) {
+            if(parameters.containsKey(PARAMETERS)) {
+                parameters.remove(PARAMETERS);
+            }
+            /** TODO:
+             * when numIds is too small, NSG graph would core/throw exception
+             * because There are too much invalid entries in the knn graph.
+             */
+            if(pair.docs.length < 256) {
+                parameters.put(INDEX_DESCRIPTION_PARAMETER, FAISS_FLAT_DESCRIPTION);
+            }
+        }
+        log.debug(String.format("parameters:[%s]",parameters.toString()));
```

Comment:
lets add some more context in this debug statement.

Reply:
as the following explain for the less document scenario in NSG graph, i try to use `FLAT` map to solve the exception. so i added debug info to see whether  changed the parameter